vega_sim/service.py

```diff
@@ -1528,3 +1533,288 @@ def get_trades(
+                Side of order, "SIDE_BUY" or "SIDE_SELL".
+            order_type (Optional[Union[vega_protos.vega.Order.Type, str]]):
+                Type of order, "TYPE_MARKET" or "TYPE_LIMIT".
+            time_in_force (Optional[Union[vega_protos.vega.Order.TimeInForce, str]]):
+                Time in force of order, determines how long order remains active.
+            price (Optional[float]):
+                Price of order, not required for "TYPE_LIMIT" orders. Defaults to None.
+            expires_at (Optional[int]):
+                Determines timestamp at which order expires, only required for orders of
+                "TYPE_LIMIT" and "TIME_IN_FORCE_GTT". Defaults to None.
+            reference (Optional[str]):
+                Reference to assign to order. Defaults to None.
+            pegged_reference (Optional[str]):
+                Reference for price offset for order. Defaults to None.
+            pegged_offset (Optional[float]):
+                Value for price offset from reference for order. Defaults to None.
+
+        Returns:
+            OrderSubmission:
+                The created Vega OrderSubmission object
+        """
+
+        price = (
+            price
+            if price is None
+            else str(
+                num_to_padded_int(
+                    to_convert=price, decimals=self.market_price_decimals[market_id]
+                )
+            )
+        )
+        pegged_offset = (
+            pegged_offset
+            if pegged_offset is None
+            else str(
+                num_to_padded_int(
+                    to_convert=pegged_offset,
+                    decimals=self.market_price_decimals[market_id],
+                )
+            )
+        )
+        size = (
+            size
+            if size is None
+            else num_to_padded_int(
+                to_convert=size, decimals=self.market_pos_decimals[market_id]
+            )
+        )
+        if (pegged_offset is not None) and (pegged_reference is not None):
+            pegged_order = trading.build_pegged_order(
+                pegged_offset=pegged_offset,
+                pegged_reference=pegged_reference,
+            )
+        else:
+            pegged_order = None
+
+        return trading.order_submission(
+            data_client=self.trading_data_client,
+            market_id=market_id,
+            price=price,
+            size=size,
+            side=side,
+            time_in_force=time_in_force,
+            expires_at=expires_at,
+            order_type=order_type,
+            reference=reference,
+            pegged_order=pegged_order,
+        )
+
+    def submit_instructions(
+        self,
+        wallet_name: str,
+        key_name: Optional[str] = None,
+        cancellations: Optional[List[OrderCancellation]] = None,
+        amendments: Optional[List[OrderAmendment]] = None,
+        submissions: Optional[List[OrderSubmission]] = None,
+    ):
```

Comment:
It feels like it would be nice to also have an interface into this method where one doesn't have to first separately create the individual objects, but I also can't think how to do that without becoming a horrible mess, so I'm open to suggestions and this one seems good too 😄

Reply:
Hm yeah I completely agree I just couldn't think of an elegant way around it either! The way the MMs are setup now it just seemed to make sense to create the objects as you pass through the loop in `_move_side`.

We could definitely add some simpler methods for other agents to use; e.g. batch_submit_orders (pass a list of prices and volumes to submit limit orders at), batch_cancel_orders (pass a list of order ids to cancel)?